internal/pkg/deploy/cloudformation/project.go

```diff
@@ -0,0 +1,145 @@
+	return &projectStackConfig{
+		Project: input,
+		box:     box,
+	}
+}
+
+// Template returns the environment CloudFormation template.
+func (e *projectStackConfig) Template() (string, error) {
+	template, err := e.box.FindString(projectTemplatePath)
+	if err != nil {
+		return "", &ErrTemplateNotFound{templateLocation: projectTemplatePath, parentErr: err}
+	}
+	return template, nil
+}
+
+// ResourceTemplate generates a StackSet template with all the Project-wide resources (ECR Repos, KMS keys, S3 buckets)
+func (e *projectStackConfig) ResourceTemplate(config *projectResourcesConfig) (string, error) {
+	stackSetTemplate, err := e.box.FindString(projectResourcesTemplatePath)
+	if err != nil {
+		return "", &ErrTemplateNotFound{templateLocation: projectResourcesTemplatePath, parentErr: err}
+	}
+
+	template, err := template.New("resourcetemplate").Parse(stackSetTemplate)
+	if err != nil {
+		return "", err
+	}
+	// Sort the account IDs and Apps so that the template we generate is deterministic
+	sort.Strings(config.Accounts)
+	sort.Strings(config.Apps)
+
+	var buf bytes.Buffer
+	if err := template.Execute(&buf, config); err != nil {
+		return "", err
+	}
+
+	return string(buf.Bytes()), nil
+}
+
+// Parameters returns the parameters to be passed into a environment CloudFormation template.
+func (e *projectStackConfig) Parameters() []*cloudformation.Parameter {
+	return []*cloudformation.Parameter{
+		{
+			ParameterKey:   aws.String(projectAdminRoleParamName),
+			ParameterValue: aws.String(e.stackSetAdminRoleName()),
+		},
+		{
+			ParameterKey:   aws.String(projectExecutionRoleParamName),
+			ParameterValue: aws.String(e.StackSetExecutionRoleName()),
+		},
+	}
+}
+
+// Tags returns the tags that should be applied to the project CloudFormation stack.
+func (e *projectStackConfig) Tags() []*cloudformation.Tag {
+	return []*cloudformation.Tag{
+		{
+			Key:   aws.String(projectTagKey),
+			Value: aws.String(e.Project.Name),
+		},
+	}
+}
+
+// StackName returns the name of the CloudFormation stack (based on the project name).
+func (e *projectStackConfig) StackName() string {
+	return fmt.Sprintf("%s-infrastructure-roles", e.Project.Name)
+}
+
+// StackSetName returns the name of the CloudFormation StackSet (based on the project name).
+func (e *projectStackConfig) StackSetName() string {
+	return fmt.Sprintf("%s-infrastructure", e.Project.Name)
+}
+
+// StackSetDescription returns the description of the StackSet for project resources.
+func (e *projectStackConfig) StackSetDescription() string {
+	return "ECS CLI Project Resources (ECR repos, KMS keys, S3 buckets)"
+}
+
+func (e *projectStackConfig) stackSetAdminRoleName() string {
+	return fmt.Sprintf("%s-adminrole", e.Project.Name)
+}
+
+// StackSetAdminRoleARN returns the role ARN of the role used to administer the Project
+// StackSet.
+func (e *projectStackConfig) StackSetAdminRoleARN() string {
+	//TODO find a partition-neutral way to construct this ARN
```

Comment:
What does this mean?

Reply:
right now the partition of the role is hard coded ("aws"). If we're going to spin this up in gov-cloud or aws-cn, this won't work (but stacksets aren't supported there yet, anyway, so we have some time).